### ISSUE TITLE
Train yolo on mot dataset

### ISSUE THREAD

**Issue by rslim97** [mikel-brostrom/yolo_tracking#1307]
### Search before asking

- [X] I have searched the Yolo Tracking [issues](https://github.com/mikel-brostrom/yolo_tracking/issues) and found no similar bug report.

### Question

Hi Mike, beginner question here. I'm wondering if it is possible to train to improve yolov7 or even yolov5 on the MOT16, MOT17 dataset or a custom one? is there any guidelines that I should follow (preferably without relying on the roboflow framework)? thanks for your suggestions. The baseline model has difficulty detecting heads of the occluded persons.

**Comment by mikel-brostrom** [comment]
For a SOTA people detector I would recommend you to follow: https://github.com/ifzhang/ByteTrack#data-preparation. WItht eh prepared data you can then easily use Yolov5 or 7 for training

**Comment by rslim97** [comment]
Hi Mike I have ran the mot to coco format conversion. I know have the annotations folder containing some .json files. What should I do next? do I need to convert the dataset format from coco to yolo?

**Comment by rslim97** [comment]
Hi Mike, I've just managed to start the training meanwhile I noticed that the objectness score and cls loss has downward trend and the mAP sometimes goes up and down but always stuck around 0.74 with the highest currently attained at epoch 2 with 0.77 but higher obj and cls loss. Is there anything to be made out of this?

**Comment by mikel-brostrom** [comment]
Did you managed a successful training @rslim97 ?

**Comment by rslim97** [comment]
@mikel-brostrom yes, I managed to train yolov7 for 300 epochs (around 26 hrs). The final weights converged to only 0.729 mAP. The best ckpt has about 0.748 mAP. I've tried on my dataset and the last weights performed worse (a lot of obvious missed detections), however sadly the best weights also just has a very slight improvement. I think the model has overfitted to mot dataset and unfortunately is not very suitable for my use case (has more close range view of people).
![photo_2024-03-08_18-07-18](https://github.com/mikel-brostrom/yolo_tracking/assets/29839184/c32752ac-db94-409d-b37b-31fa59106f46)

![Screenshot from 2024-03-06 17-22-37](https://github.com/mikel-brostrom/yolo_tracking/assets/29839184/0bea9a7c-9a00-4db3-a702-876216c4bf4c)

**Comment by mikel-brostrom** [comment]
Add more datasources to your training data so that your model generalizes better: [MOT17](https://motchallenge.net/), [MOT20](https://motchallenge.net/), [CrowdHuman](https://www.crowdhuman.org/), [Cityperson](https://github.com/Zhongdao/Towards-Realtime-MOT/blob/master/DATASET_ZOO.md), [ETHZ](https://github.com/Zhongdao/Towards-Realtime-MOT/blob/master/DATASET_ZOO.md)
